test/testing/test_services.py

```diff
@@ -31,6 +35,23 @@ def reset_mock():
     handle_event.reset_mock()
 
 
+@pytest.yield_fixture
+def spawn_thread():
+
+    threads = []
+
+    def spawn(fn, *args):
+        threads.append(eventlet.spawn(fn, *args))
+
+    yield spawn
+
+    for gt in threads:
+        try:
+            gt.kill()
+        except:
```

Comment:
does this sometimes bubble `GreenletException`? probably want `except Exception:`